src/modules/kvs/kvs.c

```diff
@@ -729,8 +729,9 @@ static void get_request_cb (flux_t *h, flux_msg_handler_t *w,
          * Otherwise, use the current root.
          */
         if (root_dirent) {
-            if (j_dirent_validate (root_dirent) < 0
-                || !(root_ref = json_object_get (root_dirent, "DIRREF"))) {
+            if (treeobj_validate (root_dirent) < 0
+                || treeobj_is_dirref (root_dirent) < 0
```

Comment:
I was reviewing each commit and going back and forth to the final source.  I guess I missed that this one was fix later - sorry about that.